apps/hotel/src/directives/btn-hotel.directive.ts

```diff
@@ -0,0 +1,92 @@
+import { Directive, ElementRef, Input, OnChanges, OnDestroy } from '@angular/core';
+import { HotelEntity } from '@contler/entity';
+import { AuthService } from '../services/auth.service';
+import { take } from 'rxjs/operators';
+import { FormGroup } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+@Directive({
+  selector: '[contlerBtnHotel]',
+})
+export class BtnHotelDirective implements OnChanges, OnDestroy {
+  @Input() contlerBtnHotel: 'primary' | 'second' | '' = 'primary';
+  /**
+   * Hay botones que no siempre llevaran el color definido por el sistema, si no que están condicionados al estado
+   * del formulario al que pertenecen, pasar el formulario y la directiva define el color correspondiente
+   * dependiente del estado
+   */
+  @Input() formValidation: FormGroup = null;
```

Comment:
```suggestion
  @Input() disabled: booleanl;
```
maybe if you catch the disable property your logic would be easier